npc/re/merchants/enchan_illusion_16_2.txt

```diff
@@ -613,3 +614,695 @@ OnProgress:
+		mes "<ITEM>" + .@reward_name$[1] + "<INFO>" + .@reward_id[1] + "</INFO></ITEM>";
+		mes "<ITEM>" + .@reward_name$[2] + "<INFO>" + .@reward_id[2] + "</INFO></ITEM>";
+		next;
+		mes "[Illusion Stone Researcher]";
+		mes "Make sure ^4d4dffyour equipment is refined to at least +9^000000 before bringing it to me.";
+		mes "Make sure you're ^4d4dffequipped with the item that you want to improve^000000.";
+		next;
+		mes "[Illusion Stone Researcher]";
+		mes "As you may have guessed, your equipment will transform into something new after this.";
+		mes "In other words, ^4d4dffit'll lose its current Refining and Upgrade levels.^000000";
+		mes "Are you ready?";
+		next;
+		.@s = select( .@reward_name$[0], .@reward_name$[1], .@reward_name$[2], "Cancel." ) - 1;
+		switch(.@s) {
+		case 0:		// Herald_Of_GOD_IL
+			.@eq_req = 2128;		// Herald_Of_GOD_
+			.@part = EQI_HAND_L;
+
+			setarray .@items_req[0],
+				25271,100,			// IllusionStone
+				25298,300;			// SpritJewel
+			break;
+		case 1:		// Clack_Of_Servival_IL
+			.@eq_req = 2509;		// Clack_Of_Servival
+			.@part = EQI_GARMENT;
+
+			setarray .@items_req[0],
+				25271,100,			// IllusionStone
+				25297,200;			// Frozen_PieceOfRock
+			break;
+		case 2:		// Cap_IL
+			.@eq_req = 2227;		// Cap_
+			.@part = EQI_HEAD_TOP;
+
+			setarray .@items_req[0],
+				25271,50,			// IllusionStone
+				25297,100;			// Frozen_PieceOfRock
+			break;
+		case 3:
+			end;
+		}
+		break;
+	}
+	mes "[Illusion Stone Researcher]";
+	mes "For that equipment, I need the following materials.";
+	mes "^4d4fff +9 " + getitemname(.@eq_req) + "";
+	mes "" + .@items_req[1] + " " + getitemname(.@items_req[0]) + "";
+	mes "" + .@items_req[3] + " " + getitemname(.@items_req[2]) + "^000000";
+	mes "Do you want to continue?";
+	next;
+	if (select( "Continue.", "I'll bring those materials." ) == 2) {
+		mes "[Illusion Stone Researcher]";
+		mes "Come back when you're ready.";
+		close;
+	}
+	if (countitem(.@items_req[0]) < .@items_req[1] || countitem(.@items_req[2]) < .@items_req[3]) {
+		mes "[Illusion Stone Researcher]";
+		mes "You don't have all the materials.";
+		close;
+	}
+	.@equip_id = getequipid(.@part);
+	if (.@equip_id == -1) {
+		mes "[Illusion Stone Researcher]";
+		if (.@part == EQI_HAND_R)
+			mes "Where's your weapon?";
+		else
+			mes "Why don't you go wear that equipment first?";
+		close;
+	}
+	if (.@equip_id != .@eq_req) {
+		mes "[Illusion Stone Researcher]";
+		mes "Please equip a ^4d4fff+9 " + getitemname(.@eq_req) + ".^000000";
+		close;
+	}
+	if (getequiprefinerycnt(.@part) < 9) {
+		mes "[Illusion Stone Researcher]";
+		mes "I can only upgrade equipment that is ^4d4fffrefined to at least +9.^000000";
+		close;
+	}
+	mes "[Illusion Stone Researcher]";
+	mes "There you go.";
+	delequip .@part;
+	delitem .@items_req[0], .@items_req[1];
+	delitem .@items_req[2], .@items_req[3];
+	getitem .@reward_id[.@s],1;
+	close;
+}
+
+
+// illusion enhancing - enhance the illusion gears
+// note: unknown enhance rate
```

Comment:
On the official **bRO** server it mentions the rates.
https://browiki.org/wiki/Cavernas_Ilusionais